### DISCUSSION TITLE
Google Tag Manager dataLayer

### DISCUSSION THREAD

**Discussion by a2mbs** [add2cal/add-to-calendar-button#606] [Q&A]
On my project, google tag manager data is not stored on window.dataLayer.  Is there any way for me to specify a different window  object to capture the google tag manager data? For example, instead of window.dataLayer use window.googDataLayer.  We use window.dataLayer for a different tracking tool.  I tried using mutation observer for tracking 'atcb-last-event', however, it does not tell me which option user selects, but the google dataLayer does as I've observed in the demos.  Ultimately, I'd like the 'atcb-last-event' to show the same events as in google tracking.

**Comment by jekuer** [comment]
Regarding "atcb-last-event": It basically does show the same data. However, the "open" events happen more or less at the same time as the "success" event. While at the dataLayer, you can see the history, you cannot here as it would blow up the markdown.

Making the window element for tracking configurable is something we definitely will think about. However, it could be tricky to track things like "initialization" when we first need to parse the config - this is somehow contradictive.
Not fully sure about a valid, secure, and good solution here.

Can't you just simply also use the dataLayer in GTM, but only pull out the event data from the button?

**Comment by a2mbs** [comment]
So, for us, dataLayer is not used for GTM.  GTM is set to use window.gtagLayer.  In fact, the code that atcb has to push to window.dataLayer actually doesn't work in our case as our window.dataLayer doesn't even have a push function.

I will try some other options if there's no other way you can think of.
